provider/clusterrosa/hcp/resource.go

```diff
@@ -0,0 +1,1455 @@
+	clusterWait       common.ClusterWait
+}
+
+var _ resource.ResourceWithConfigure = &ClusterRosaHcpResource{}
+var _ resource.ResourceWithImportState = &ClusterRosaHcpResource{}
+
+func New() resource.Resource {
+	return &ClusterRosaHcpResource{}
+}
+
+func (r *ClusterRosaHcpResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cluster_rosa_hcp"
+}
+
+func (r *ClusterRosaHcpResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "OpenShift managed cluster using rosa sts.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Unique identifier of the cluster.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					// This passes the state through to the plan, preventing
+					// "known after apply" since we know it won't change.
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"external_id": schema.StringAttribute{
+				Description: "Unique external identifier of the cluster. " + common.ValueCannotBeChangedStringDescription,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "Name of the cluster. Cannot exceed 15 characters in length. " + common.ValueCannotBeChangedStringDescription,
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtMost(15),
+				},
+			},
+			"cloud_region": schema.StringAttribute{
+				Description: "Cloud region identifier, for example 'us-east-1'.",
+				Required:    true,
+			},
+			"sts": schema.SingleNestedAttribute{
+				Description: "STS configuration.",
+				Attributes:  sts.HcpStsResource(),
+				Optional:    true,
+			},
+			"properties": schema.MapAttribute{
+				Description: "User defined properties.",
+				ElementType: types.StringType,
+				Optional:    true,
+				Computed:    true,
+				Validators:  []validator.Map{rosa.PropertiesValidator},
+			},
+			"ocm_properties": schema.MapAttribute{
+				Description: "Merged properties defined by OCM and the user defined 'properties'.",
+				ElementType: types.StringType,
+				Computed:    true,
+			},
+			"tags": schema.MapAttribute{
+				Description: "Apply user defined tags to all cluster resources created in AWS. " + common.ValueCannotBeChangedStringDescription,
+				ElementType: types.StringType,
+				Optional:    true,
+			},
+			"etcd_encryption": schema.BoolAttribute{
+				Description: "Encrypt etcd data. Note that all AWS storage is already encrypted. " + common.ValueCannotBeChangedStringDescription,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.UseStateForUnknown(),
+				},
+			},
+			// "autoscaling_enabled": schema.BoolAttribute{
```

Comment:
I removed it because the autoscaling is within the node pool resource this time around. Let me remove it entirely

Reply:
I'd rather have the control for this directly on the node pool resource instead of the cluster level. Trying to move away from managing it through the cluster